deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,303 @@
+    ) external;
+}
+
+interface RolesLike {
+    function setIlkAdmin(bytes32, address) external;
+    function setUserRole(bytes32, address, uint8, bool) external;
+    function setRoleAction(bytes32, uint8, address, bytes4, bool) external;
+}
+
+interface RegistryLike {
+    function file(bytes32, bytes32, address) external;
+}
+
+interface VaultLike {
+    function ilk() external view returns (bytes32);
+    function roles() external view returns (address);
+    function buffer() external view returns (address);
+    function vat() external view returns (address);
+    function nst() external view returns (address);
+    function file(bytes32, address) external;
+    function draw(uint256) external;
+    function wipe(uint256) external;
+}
+
+interface BufferLike {
+    function approve(address, address, uint256) external;
+}
+
+interface SwapperLike {
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function swap(address, address, uint256, uint256, address, bytes calldata) external returns (uint256);
+}
+
+interface DepositorUniV3Like {
+    struct LiquidityParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+        uint128 liquidity;
+        uint256 amt0Desired;
+        uint256 amt1Desired;
+        uint256 amt0Min;
+        uint256 amt1Min;
+    }
+
+    struct CollectParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+    }
+
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function deposit(LiquidityParams memory) external returns (uint128, uint256, uint256);
+    function withdraw(LiquidityParams memory, bool) external returns (uint128, uint256, uint256, uint256, uint256);
+    function collect(CollectParams memory) external returns (uint256, uint256);
+}
+
+interface StableSwapperLike {
+    function swapper() external view returns (address);
+}
+
+interface StableDepositorUniV3Like {
+    function depositor() external view returns (address);
+}
+
+interface ConduitMoverLike {
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+}
+
+interface KissLike {
+    function kiss(address) external;
+}
+
+struct AllocatorIlkConfig {
+    bytes32 ilk;
+    uint256 duty;
+    uint256 debtCeiling;
+    address allocatorProxy;
+    uint8 facilitatorRole;
+    uint8 automationRole;
+    address[] facilitators;
+    address[] stableSwapperKeepers;
+    address[] stableDepositorUniV3Keepers;
+    address[] conduitMoverKeepers;
+    address[] swapTokens;
+    address[] depositTokens;
+    address ilkRegistry;
+}
+
+function bytes32ToStr(bytes32 _bytes32) pure returns (string memory) {
```

Comment:
The problem with this implementation is that the resulting `string` will always be of length 32 (right-padded with empty bytes). But other ilks in the `IlkRegistry` have names and symbols whose string length matches the number of characters (e.g. "WETH" is a string of length 4 and not 32).

Would suggest this alternative implementation instead:
```
        uint256 len;
        while(len < 32 && _bytes32[len] != 0) len++;
        bytes memory bytesArray = new bytes(len);
        for (uint256 i; i < len; i++) {
            bytesArray[i] = _bytes32[i];
        }
        return string(bytesArray);
```

Reply:
Interesting, I actually took that function from the ilk registry code:
https://github.com/makerdao/ilk-registry/blob/master/src/IlkRegistry.sol#L429
It's result might have never really been used, since it's only used in add() in a scenario the gem is missing a name or a symbol. For special ilks like RWAs put() was used (which got the name has input from the spell).

Reply:
The one you suggested is better in this case.
Changed.